### PR TITLE
Keys.SKULL_TYPE not working

### PR DESCRIPTION
**Pull request by rexbut**
@Aaron1011 already fixed the problem in commit https://github.com/SpongePowered/SpongeCommon/commit/8970a6a88b44ebba77d5752d8e4f1f0fde98e781 but commit https://github.com/SpongePowered/SpongeCommon/commit/9d594880e16905f61784f37d0bc890da05981caf canceled the fix.

This also solves problem https://github.com/SpongePowered/SpongeCommon/issues/1406

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/processor/common/SkullUtils.java [hunk 1]
```diff
@@ -57,7 +57,9 @@ public static boolean supportsObject(Object object) {
     }
 
     public static SkullType getSkullType(int skullType) {
-        return Iterables.get(SpongeImpl.getRegistry().getAllOf(SkullType.class), skullType);
+        return SpongeImpl.getRegistry().getAllOf(SkullType.class).stream()
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/processor/common/SkullUtils.java]: I almost want to say that I'd rather a traditional for loop to a stream, since streams are not quite yet optimized. Given the potential requirement of this being considered a "hot" method in some cases, it's best to be fast than to use fancy new API's, at least in the current implementation with streams.

**Review by gabizou — CHANGES_REQUESTED** [review]

**Comment by ryantheleach** [comment]
This doesn't seem to solve #1406 (in my dev environment, may be an error with fetching skulls from mojang though)

Nor did the preexisting code break skulls?

https://gist.github.com/ryantheleach/422e144331286524f3b746a9ec6ad4ef

What testing procedures did you use @rexbut ?

**Comment by ryantheleach** [comment]
**Edit: Ignore this, it looks good. my debugging was only getting called for the defaults for the other skulls.**

> This fixes #1406
> 
> But it didn't seem to have any difference for the other skull types, vs the previous code.
> 
> https://gist.github.com/ryantheleach/422e144331286524f3b746a9ec6ad4ef
> 
> What testing procedures did you use @rexbut ?
> 
> So both the original and the new version, whilst they work, still seem incorrect.
> 
> I would have been expecting the id's to be fixed.

**Comment by ryantheleach** [comment]
CI is failing due to bleeding not being updated in rexbut/SpongeAPI

Code works well against the test plugin I made, but the test plugin might need style tweaks? I kind of rushed it a little near the end.

**Review by parlough — APPROVED** [review]

File: testplugins/src/main/java/org/spongepowered/test/SkullTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            .extendedDescription(Text.of("commands:\n", "Give you a skull"))
+            .arguments(playerOrSource(PLAYER), GenericArguments.integer(SKULL))
+            .executor(SkullTest::giveSkull)
+            .build();
+    }
+
+    private static CommandResult giveSkull(CommandSource commandSource, CommandContext commandContext) {
+        Player target = (Player) commandContext.getOne(PLAYER).get();
+        int skulltype = (int) commandContext.getOne(SKULL).get();
+        SkullType[] skulls = {
```
Comment (gabizou) [inline testplugins/src/main/java/org/spongepowered/test/SkullTest.java]: Why are you using integers? Why not just use the actual catalog id that we provide, and the command would be able to parse correctly? It would avoid the issue of figuring out invalid integers.

**Review by gabizou — CHANGES_REQUESTED** [review]
Get rid of these magic numbers...

File: testplugins/src/main/java/org/spongepowered/test/SkullTest.java (hunk 2 above)
Comment (ryantheleach) [reply testplugins/src/main/java/org/spongepowered/test/SkullTest.java]: It went through a few iterations, I was also wanting an additional option, (that could have been satisfied by a sub-command ) to spawn a head from an unknown game profile.

The false reasoning (since edits happened) behind using int's is that I was originally going to hook straight into Skull Utils to test, then decided that I'd use the SkullTypes as registered.

The reason for less validation, was if it had of hooked into SkullUtils direct, it would have been extensible for future testing, should any skulls be introduced that wern't in the registry.

But that also ended up void, as SkullUtils didn't get edited to use internal values and not the registry.

I didn't like the potential of having wrong ints in the registry, end up working somehow, because it happened to be wrong twice.

Basically, I wrote the tests first, didn't revisit, I'll change it in 8 hours if rexbut doesn't beat me to it.

**Comment by ryantheleach** [comment]
@gabizou Is this good to go now?

**Review by parlough — APPROVED** [review]
